travel.py

```diff
@@ -182,14 +337,69 @@ def login(tenant):
         except NetworkException:
             print("Network error received - is Couchbase Server running on this host?")
         else:
-            return jsonify({'data': {'token': genToken(user)}, 'context': querytype + user})
+            return jsonify({'data': {'token': genToken(user)}, 'context': [querytype + user]})
 
         return abortmsg(401, "Failed to get user data")
 
     @api.route('/tenants/<tenant>/user/signup', methods=['POST', 'OPTIONS'])
     @cross_origin(supports_credentials=True)
     def signup(tenant):
-        """Signup a new user"""
+        """Signup a new user
+        ---
+        tags:
+        - tenants
+        parameters:
+            - name: tenant
+              in: path
+              required: true
+              schema:
+                type: string
+              example: tenant_agent_00
+              description: Tenant agent name
+        requestBody:
+            content:
+              application/json:
+                schema:
+                  type: object
+                  required:
+                   - user
+                   - password
+                  properties:
+                    user:
+                      type: string
+                      example: "user1"
+                    password:
+                      type: string
+                      example: "password1"
+        responses:
+            200:
+              description: Returns login data and query context information
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      context:
+                        type: array
+                        format: string
+                        example: ["A description of a KV operation"]
+                      data:
+                        type: object
+                        properties:
+                          token:
+                            type: string
+                            example: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoibXNfdXNlciJ9.GPs8two_vPVBpdqD7cz_yJ4X6J9yDTi6g7r9eWyAwEM
+            409:
+              description: Returns a conflict error
+              content:
+                application/json:
+                    schema:
+                       type: object
+                       properties:
+                        message:
+                           type: string
+                           example: ["User already exists"]
+        """
```

Comment:
OK, so seeing these ~50 lines repeated from above does actually motivate figuring out the `$ref` stuff doesn't it? 😂

Reply:
indeed, so hard to keep up with all of this comment, indentation hell :D